src/playbin.js

```diff
@@ -0,0 +1,157 @@
+        this._videocrop = this._playbin.videoFilter.get_by_name('videocrop');
+
+        this._video_widget = this._playbin.videoSink.widget;
+        this._video_widget.expand = true;
+        this._video_widget.noShowAll = true;
+        this._video_widget.ignoreAlpha = false;
+        this._video_widget.hide();
+        this.add(this._video_widget);
+
+        this._video_widget.connect('size-allocate', this._updateCropArea.bind(this));
+
+        this._playbin.get_bus().add_watch(0, this._bus_watch.bind(this));
+    }
+
+    _updateCropArea () {
+        const width = this._video_widget.get_allocated_width();
+        const height = this._video_widget.get_allocated_height();
+
+        if (width < 2 || height < 2 || !this._video_width || !this._video_height)
+            return;
+
+        const crop_x = (this._video_width - width) / 2;
+        const crop_y = (this._video_height - height) / 2;
+
+        this._videocrop.left = crop_x;
+        this._videocrop.right = crop_x;
+        this._videocrop.top = crop_y;
+        this._videocrop.bottom = crop_y;
+    }
+
+    _updateSizeFromStreamCollection (collection) {
+        const n = collection.get_size();
+        let stream = null;
+
+        /* Find video stream */
+        for (let i = 0; i < n; i++) {
+            stream = collection.get_stream(i);
+
+            if (stream.get_stream_type() === Gst.StreamType.VIDEO)
+                break;
+        }
+
+        if (!stream)
+            return;
+
+        /* Get video size */
+        const caps = stream.get_caps();
+        const caps_struct = caps.get_structure(0);
+        const [width_ok, width] = caps_struct.get_int('width');
+        const [height_ok, height] = caps_struct.get_int('height');
+
+        if (width_ok && height_ok) {
+            this._video_width = width;
+            this._video_height = height;
+            this._updateCropArea();
+        }
+    }
+
+    _bus_watch (bus, msg) {
+        if (msg.type === Gst.MessageType.EOS) {
+            this.emit('done');
+        } else if (msg.type === Gst.MessageType.STREAMS_SELECTED) {
+            this._updateSizeFromStreamCollection(msg.parse_streams_selected());
+        } else if (msg.type === Gst.MessageType.STATE_CHANGED) {
+            const [, newstate] = msg.parse_state_changed();
+
+            if (!this._started && newstate === Gst.State.PLAYING) {
+                this._video_widget.show_now();
+                this._started = true;
+                this.emit('start');
+            }
+        }
+
+        return GLib.SOURCE_CONTINUE;
+    }
+
+    setBackground (url) {
+        try {
+            this._css_provider.load_from_data(`
+playbin.locked {
+    background: url('${url}') no-repeat center;
+}`
+            );
+        } catch (e) {
+            logError(e, 'Failed to load css');
+        }
+    }
+
+    add_class (klass) {
+        this._style.add_class(klass);
+    }
+
+    remove_class (klass) {
+        this._style.remove_class(klass);
+    }
```

Comment:
I don't find it worth it to have these special methods here, since the playbin is a widget itself. Clients can simply call `get_style_context()` on it directly as they would on other widgets.